### PR TITLE
CAL-173 Add thread locking to UDP video processing

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Add thread locking around UDP buffer processing and improve exception logging when processing UDP packets.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining 
@jlcsmith 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire
#### How should this be tested?
Run within a window VM and stream with FFMPEG.
#### What are the relevant tickets?

[CAL-173](https://codice.atlassian.net/browse/CAL-173)
#### Checklist:
- [ ] Documentation Updated
	- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java [hunk 1]
```diff
@@ -143,37 +147,41 @@ protected void decode(ChannelHandlerContext ctx, DatagramPacket msg, List<Object
         notNull(msg, "msg must be non-null");
         notNull(outputList, "outputList must be non-null");
 
-        checkSecuritySubject(msg);
+        LOCK.lock();
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java]: Do we need to lock this entire block?  If the critical section is just writing to the buffer, can we just lock that line?

Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java]: Any that modifies the buffer, which includes reads and writes (reads modify the buffer state), needs to be protected. The only line that doesn't have to be protected is the checkSecuritySubject. I could change that one line of you want.

Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java]: No, I didn't realize the read needed to be protected

**Comment by bdeining** [comment]
Hero Build was successful, tested on Windows with FFMPEG 👍

**Comment by glenhein** [comment]
BTW: If this solution is accepted, then I need to spend a little time cleaning up the stream generator code.

File: distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java [hunk 2]
```diff
@@ -129,11 +176,17 @@ public static void main(String[] args) {
...
+                port,
+                tsDurationMillis,
+                minDatagramSize,
+                maxDatagramSize,
+                fractionalTs);
     }
 
     public static void broadcastVideo(String videoFilePath, String ip, int port,
-            long tsDurationMillis) {
+            long tsDurationMillis, int minDatagramSize, int maxDatagramSize, boolean fractionalTs) {
```
Comment (bdeining) [inline distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java]: VideoTest.java in the itests uses this method and will also have to be updated

Comment (glenhein) [inline distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java]: fixed

**Pull request merged**